### PR TITLE
Fix crash in ReportStatsForBuildSummary when execute phase doesn't exit cleanly

### PR DESCRIPTION
**Pull request by mpysson**

### CODE REVIEW

File: cg/nuget/cgmanifest.json [hunk 1]
```diff
@@ -2670,7 +2670,7 @@
         "Type": "NuGet",
         "NuGet": {
           "Name": "runtime.osx-x64.BuildXL",
-          "Version": "1.99.99"
+          "Version": "2.0.1"
```
Comment (mpysson) [inline cg/nuget/cgmanifest.json]: @rijulluman-msft - this is unrelated to my change. Is it just due to staging of when we enabled the enforcement side of things?

Comment (Kr1sso) [reply cg/nuget/cgmanifest.json]: I haven't been in the loop about this - what does this indicate, cause if its not updated?

Comment (mpysson) [reply cg/nuget/cgmanifest.json]: It's automatically generated in dev builds. Soon, PR and rolling builds will fail if the cgmanifest file is out of sync with the packages that are fetched. The purpose of the file is to pump package information into the component governance system.

Comment (almiliMSFT) [reply cg/nuget/cgmanifest.json]: yes, this is exactly because the cgmanifest file checked in with Rijul's change was a bit stale

Comment (Kr1sso) [reply cg/nuget/cgmanifest.json]: Thanks!

Comment (rijulluman-msft) [reply cg/nuget/cgmanifest.json]: Yes, looks like the file I generated was a bit stale
